### PR TITLE
Maui

### LINKED ISSUE #34714: Package Request: mauikit
[MauiKit](https://mauikit.org/) is a set of templated controls and tools based on QQC2 and Kirigami shared among the Maui set of applications. MauiKit helps to quickly build UIs that follow the Maui HIG and bring ready-to-go tools for different platforms, such as Android and Linux.

A 'devel' package would also be needed for development purposes. Also, [mauikit apps](https://invent.kde.org/maui) (like station, nota, buho, etc) needs to be packaged separately. I will raise a separate issues for these if you want me to.

### PR DESCRIPTION
**Pull request by Idesmi**
WIP for the Maui collection of framework and applications

Closes #34714

### CODE REVIEW

**Comment by paper42** [comment]
Add this to common/shlibs:
```
libMauiKit.so mauikit-2.1.1_1
libMauiKitFileBrowsing.so.2 mauikit-filebrowsing-2.1.1_1
```

File: srcpkgs/maui-shell/template [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Template file for 'maui-shell'
+pkgname=maui-shell
+version=0.5.0
+revision=1
+build_style=cmake
+hostmakedepends="extra-cmake-modules gettext pkg-config"
+makedepends="qt5-declarative-devel qt5-devel qt5-svg-devel qt5-quickcontrols2-devel qt5-wayland-devel ki18n-devel kconfig-devel kservice-devel mauikit-devel mauikit-filebrowsing-devel krunner-devel knotifications-devel knotifyconfig-devel kdesu-devel kwallet-devel kidletime-devel kdeclarative-devel kglobalaccel-devel kwayland-devel kpeople-devel kactivities5-stats-devel kactivities5-devel kio-devel prison-devel kpackage-devel kiconthemes-devel kunitconversion-devel kitemmodels-devel kinit-devel ktexteditor-devel kdoctools-devel kcrash-devel kdbusaddons-devel kcoreaddons-devel karchive-devel fontconfig-devel phonon-qt5-devel kded-devel"
```
Comment (paper42) [inline srcpkgs/maui-shell/template:7]: wrap at 80 columns

File: srcpkgs/mauikit/template [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Template file for 'mauikit'
+pkgname=mauikit
+version=2.1.1
+revision=1
+build_style=cmake
+hostmakedepends="extra-cmake-modules gettext"
+makedepends="qt5-devel qt5-declarative-devel qt5-quickcontrols2-devel qt5-svg-devel kirigami2-devel ki18n-devel kcoreaddons-devel knotifications-devel kconfig-devel kwindowsystem-devel qt5-x11extras-devel xcb-util-wm-devel"
+depends="kirigami2>=5.40.0"
```
Comment (paper42) [inline srcpkgs/mauikit/template:8]: drop this version restriction, this is guaranteed by entry in common/shlibs

**Comment by Idesmi** [comment]
> Add this to common/shlibs:
> 
> ```
> libMauiKit.so mauikit-2.1.1_1
> libMauiKitFileBrowsing.so.2 mauikit-filebrowsing-2.1.1_1
> ```

Hi, I already did locally, in order to test the build. I'm just new to git, and mean to soft reset to upstream and do again one commit per package as it's defined in the contributing guidelines.

**Comment by paper42** [comment]
> Hi, I already did locally, in order to test the build. I'm just new to git, and mean to soft reset to upstream and do again one commit per package as it's defined in the contributing guidelines.

There is a few ways to do this. This is one, but also it would be painful and not optimal. I would suggest to put your changes to each package to a new commit, then `git rebase -i upstream/master` and squash/fixup commits that should be one.

**Comment by Idesmi** [comment]
Now I'd need to add `qt5-declarative` too to hostmakedepends...
But I don't understand why can't it be found in the chroot instead.

Are there recurrent packages that need to be installed in the host to build KF5/Qt5 packages?

**Pull request closed without merging**